src/__Private/LintRunConfig.hack

```diff
@@ -53,7 +53,7 @@ final class LintRunConfig {
     >,
     // Each linter may specify a type for itself.
     // The type of this key is effectively `dict<classname<T1 ... Tn>, Tx>`
-    ?'linterConfigs' => dict<string, BaseLinter::TConfig>,
+    ?'linterConfigs' => dict<string, dynamic>,
```

Comment:
Why do we need dynamic here? Would `mixed` or `shape(...)` cause type errors?

Reply:
- `shape(...)` states that any shape is acceptable
- `mixed` indicates that any value is acceptable
- `dynamic` means that the acceptable types vary - and given we refine it at runtime, seems the most correct

Reply:
I have very little experience with the local permissive operations of dynamic. If this type was chosen for documentation ppurposes, would it be an idea to declare `type TConfigurationDiffersPerLinter = mixed;` and stick that in there. Then you can'tgget caught of guard by the typechecker not warning you about incorrect local operations, but the documentation purpose is very clear.

Reply:
The local behavior shouldn't be an issue here, as it gets handled within the internals of hhast, not the linters